ibm/service/kms/resource_ibm_kms_key.go

```diff
@@ -166,103 +166,24 @@ func ResourceIBMKmskey() *schema.Resource {
-
-	name := d.Get("key_name").(string)
-	standardKey := d.Get("standard_key").(bool)
-
-	var expiration *time.Time
-	if es, ok := d.GetOk("expiration_date"); ok {
-		expiration_string := es.(string)
-		// parse string to required time format
-		expiration_time, err := time.Parse(time.RFC3339, expiration_string)
-		if err != nil {
-			return fmt.Errorf("[ERROR] Invalid time format (the date format follows RFC 3339): %s", err)
-		}
-		expiration = &expiration_time
-	} else {
-		expiration = nil
+	key, err := kpAPI.CreateImportedKey(context.Background(), keyData.Name, keyData.Expiration, keyData.Payload, keyData.EncryptedNonce, keyData.IV, keyData.Extractable)
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error while creating key: %s", err)
 	}
 
-	var keyCRN string
-	if standardKey {
-		if v, ok := d.GetOk("payload"); ok {
-			//import standard key
-			payload := v.(string)
-			stkey, err := kpAPI.CreateImportedStandardKey(context.Background(), name, expiration, payload)
-			if err != nil {
-				return fmt.Errorf("[ERROR] Error while creating standard key with payload: %s", err)
-			}
-			keyCRN = stkey.CRN
-			d.SetId(keyCRN)
-
-		} else {
-			//create standard key
-			stkey, err := kpAPI.CreateStandardKey(context.Background(), name, expiration)
-			if err != nil {
-				return fmt.Errorf("[ERROR] Error while creating standard key: %s", err)
-			}
-			keyCRN = stkey.CRN
-			d.SetId(keyCRN)
-
-		}
-	} else {
-		if v, ok := d.GetOk("payload"); ok {
-			payload := v.(string)
-			encryptedNonce := d.Get("encrypted_nonce").(string)
-			iv := d.Get("iv_value").(string)
-			stkey, err := kpAPI.CreateImportedRootKey(context.Background(), name, expiration, payload, encryptedNonce, iv)
-			if err != nil {
-				return fmt.Errorf("[ERROR] Error while creating Root key with payload: %s", err)
-			}
-			keyCRN = stkey.CRN
-			d.SetId(keyCRN)
-
-		} else {
-			stkey, err := kpAPI.CreateRootKey(context.Background(), name, expiration)
-			if err != nil {
-				return fmt.Errorf("[ERROR] Error while creating Root key: %s", err)
-			}
-			keyCRN = stkey.CRN
-			d.SetId(keyCRN)
-		}
-	}
+	d.SetId(key.CRN)
 	return resourceIBMKmsKeyUpdate(d, meta)
 }
 
 func resourceIBMKmsKeyRead(d *schema.ResourceData, meta interface{}) error {
-	instanceCRN, instanceID, keyid := getInstanceAndKeyDataFromCRN(d.Id())
 
-	kpAPI, _, err := populateKPClient(d, meta, instanceID)
-	if err != nil {
-		return err
-	}
-	// keyid := d.Id()
-	key, err := kpAPI.GetKey(context.Background(), keyid)
-	if err != nil {
-		kpError := err.(*kp.Error)
-		if kpError.StatusCode == 404 || kpError.StatusCode == 409 {
-			d.SetId("")
-			return nil
-		}
-		return fmt.Errorf("[ERROR] Get Key failed with error while reading Key: %s", err)
-	} else if key.State == 5 { //Refers to Deleted state of the Key
-		d.SetId("")
-		return nil
-	}
-
-	err = setKeyDetails(d, meta, instanceID, instanceCRN, key, kpAPI)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	_, err := KMSKeyReadHelper(d, meta)
```

Comment:
As a standard, this function name can be renamed as a verb indicating what this function does